wit/gpio.wit

```diff
@@ -0,0 +1,38 @@
+variant gpio-error {
```

Comment:
Could you please add some docs on this new WIT file? What are each resources represent, much like the `pwm-output-pin` resource.